### PR TITLE
Test VRF v2 locally using a mock contract

### PR DESCRIPTION
**Pull request by aelmanaa**

### CODE REVIEW

File: src/pages/vrf/v2/subscription/examples/test-locally.mdx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+      <ClickToZoom src="/images/vrf/mock/v2-firstrandomword.jpg" />
+
+   1. Fill in the index with _1_ then click on _call_ to read the second random word.
+
+      {" "}
+
+      <ClickToZoom src="/images/vrf/mock/v2-secondrandomword.jpg" />
+
+## Conclusion
```
Comment (thedriftofwords) [inline src/pages/vrf/v2/subscription/examples/test-locally.mdx:158]: Nit: I haven't seen most docs tutorials end with a section actually titled "Conclusion" - my suggestion is to retitle this "Next steps" if it doesn't fit well into the usual structure for our "What's next" sections.

After making that change, we would avoid adding a "What's next" section to these specific pages to avoid duplication.

File: src/pages/vrf/v2/subscription/examples/test-locally.mdx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+1. Under _DEPLOY & RUN TRANSACTIONS_, select _VRFCoordinatorV2Mock_.
+
+   <ClickToZoom src="/images/vrf/mock/v2-deploymock.jpg" />
+
+1. Under _DEPLOY_, fill in the `_BASEFEE` and `_GASPRICELINK`. These variables are used in the _VRFCoordinatorV2Mock_ contract to represent the base fee and the gas price (in LINK tokens) for the VRF requests. You can set: `_BASEFEE=100000000000000000` and `_GASPRICELINK=1000000000`.
+
+1. Click on _transact_ to deploy the _VRFCoordinatorV2Mock_ contract.
+
+1. Once deployed, you should see the _VRFCoordinatorV2Mock_ contract under _Deployed Contacts_.
```
Comment (thedriftofwords) [inline src/pages/vrf/v2/subscription/examples/test-locally.mdx:66]:
```suggestion
1. Once deployed, you should see the _VRFCoordinatorV2Mock_ contract under _Deployed Contracts_.
```

Comment (thedriftofwords) [reply src/pages/vrf/v2/subscription/examples/test-locally.mdx:66]: Search for additional instances in this file and also apply changes to direct funding file

File: src/pages/vrf/v2/subscription/examples/test-locally.mdx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  description: "Example contract for generating random words using the VRF v2 subscription method on your local blockchain using a mock contract."
+---
+
+import VrfCommon from "@features/vrf/v2/common/VrfCommon.astro"
+import ContentCommon from "@features/common/ContentCommon.astro"
+import { CodeSample, ClickToZoom, Aside } from "@components"
+
+<VrfCommon callout="subscription" />
+
+This guide explains how to test Chainlink VRF v2 on a [Remix IDE](https://remix-ide.readthedocs.io/en/latest/run.html#environment) sandbox blockchain environment. **Note**: You can reuse the same logic on another development environment, such as hardhat or truffle. For example, read the hardhat starter kit [RandomNumberConsumer unit tests](https://github.com/smartcontractkit/hardhat-starter-kit/blob/main/test/unit/RandomNumberConsumer.spec.js).
```
Comment (thedriftofwords) [inline src/pages/vrf/v2/subscription/examples/test-locally.mdx:15]: Capitalize these terms when referring to them in plain text (not code references): 

- "Hardhat"
- "Truffle"
- "Hardhat Starter Kit"

Apply throughout this document and the direct funding version.

**Review by thedriftofwords — CHANGES_REQUESTED** [review]

File: src/features/common/local-testing-benefits.mdx [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+1. Saving your native testnet tokens: Deploying and interacting with contracts requires paying gas fees. Although testnet native tokens do not have any value associated with them, their supply is limited by public faucets. Using mock contracts locally allows you to test your contracts freely without incurring any expenses.
+1. Controlled environment: Local testing allows you to create a controlled environment where you can manipulate various parameters, such as block time and gas prices, to test your smart contracts' function as expected under different conditions.
+1. Isolated testing: You can focus on testing individual parts of your contract, ensuring they work as intended before integrating them with other components.
+1. Easier debugging: Because local tests run on your machine, you have better control over the debugging process. You can set breakpoints, inspect variables, and step through your code to identify and fix issues.
+1. Comprehensive test coverage: You can create test cases to cover all possible scenarios and edge cases.
```
Comment (dwightjl) [inline src/features/common/local-testing-benefits.mdx:8]:
```suggestion
- Faster feedback loop: Immediate feedback on the functionality and correctness of your smart contracts. This helps you quickly identify and fix issues without waiting for transactions to be mined/validated on a testnet.
- Saving your native testnet tokens: Deploying and interacting with contracts requires paying gas fees. Although testnet native tokens do not have any value associated with them, their supply is limited by public faucets. Using mock contracts locally allows you to test your contracts freely without incurring any expenses.
- Controlled environment: Local testing allows you to create a controlled environment where you can manipulate various parameters, such as block time and gas prices, to test your smart contracts' function as expected under different conditions.
- Isolated testing: You can focus on testing individual parts of your contract, ensuring they work as intended before integrating them with other components.
- Easier debugging: Because local tests run on your machine, you have better control over the debugging process. You can set breakpoints, inspect variables, and step through your code to identify and fix issues.
- Comprehensive test coverage: You can create test cases to cover all possible scenarios and edge cases.
```

Use unordered lists for items that are not procedural.

File: src/pages/vrf/v2/subscription/examples/test-locally.mdx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+1. Under _DEPLOY & RUN TRANSACTIONS_, select _RandomNumberConsumerV2_.
+
+   <ClickToZoom src="/images/vrf/mock/v2-deployconsumer.jpg" />
+
+1. Under _DEPLOY_, fill in `SUBSCRIPTIONID` with your subscription ID, `vrfCoordinator` with the deployed _VRFCoordinatorV2Mock_ address and, _KEYHASH_ with an arbitrary `bytes32` (In this example, you can set the _KEYHASH_ to `0xd89b2bf150e3b9e13446986e571fb9cab24b13cea0a43ea20a6049a85cc807cc`).
+
+1. Click on _transact_ to deploy the _RandomNumberConsumerV2_ contract.
+
+1. Once deployed, you should see the _RandomNumberConsumerV2_ contract under _Deployed Contracts_.
```
Comment (dwightjl) [inline src/pages/vrf/v2/subscription/examples/test-locally.mdx:94]:
```suggestion
1. After the consumer contract is deployed, you should see the _RandomNumberConsumerV2_ contract under _Deployed Contracts_.
```

File: src/pages/vrf/v2/subscription/examples/test-locally.mdx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+### Check the results
+
+1. Under _Deployed Contracts_, open the functions list of your deployed _RandomNumberConsumerV2_ contract.
+
+1. Click on `s_requestId` to display the last request ID. In this example, the output is _1_.
+
+   <ClickToZoom src="/images/vrf/mock/v2-deployedconsumerrequestid.jpg" />
+
+1. Each time you make a VRF request, your consumer contract requests two random words. Once the request is fulfilled, the two random words are stored in the `s_randomWords` array. You can check the stored random words by reading the two first indexes of the `s_randomWords` array. To do so, click on the _s_randomWords_ function and:
```
Comment (dwightjl) [inline src/pages/vrf/v2/subscription/examples/test-locally.mdx:144]:
```suggestion
1. Each time you make a VRF request, your consumer contract requests two random words. After the request is fulfilled, the two random words are stored in the `s_randomWords` array. You can check the stored random words by reading the two first indexes of the `s_randomWords` array. To do so, click on the _s_randomWords_ function and:
```

File: src/features/common/local-testing-benefits.mdx [hunk 7]
```diff
@@ -0,0 +1,8 @@
+Testing locally using mock contracts saves you time and resources during development. Some of the key benefits include:
```
Comment (dwightjl) [inline src/features/common/local-testing-benefits.mdx:1]: What's the reason for single-sourcing this as common? It's fine to leave it, but I only see it used in one place.

**Review by dwightjl — CHANGES_REQUESTED** [review]

File: src/features/common/local-testing-benefits.mdx (hunk 7 above)
Comment (aelmanaa) [reply src/features/common/local-testing-benefits.mdx:1]: @dwightjl it's used in 2 diffrent files:

- src/pages/vrf/v2/subscription/examples/test-locally.mdx
- src/pages/vrf/v2/direct-funding/examples/test-locally.mdx

**Review by dwightjl — APPROVED** [review]
